internal/gitprovider/github/github.go

```diff
@@ -81,10 +112,30 @@ func NewProvider(
 	return &provider{
 		owner:  owner,
 		repo:   repo,
-		client: client,
+		client: &githubClientWrapper{client},
 	}, nil
 }
 
+type githubClientWrapper struct {
+	client *github.Client
+}
+
+func (g githubClientWrapper) CreatePullRequest(ctx context.Context, owner string, repo string, pull *github.NewPullRequest) (*github.PullRequest, *github.Response, error) {
+	return g.client.PullRequests.Create(ctx, owner, repo, pull)
+}
+
+func (g githubClientWrapper) ListPullRequests(ctx context.Context, owner string, repo string, opts *github.PullRequestListOptions) ([]*github.PullRequest, *github.Response, error) {
+	return g.client.PullRequests.List(ctx, owner, repo, opts)
+}
+
+func (g githubClientWrapper) GetPullRequests(ctx context.Context, owner string, repo string, number int) (*github.PullRequest, *github.Response, error) {
+	return g.client.PullRequests.Get(ctx, owner, repo, number)
+}
+
+func (g githubClientWrapper) AddLabelsToIssue(ctx context.Context, owner string, repo string, number int, labels []string) ([]*github.Label, *github.Response, error) {
+	return g.client.Issues.AddLabelsToIssue(ctx, owner, repo, number, labels)
+}
+
```

Comment:
Instead of creating a client wrapper, could the `githubClient` interface just be the relevant subset of the methods the client already actually has? This way we can just use the existing (unwrapped) client as the implementation of the `githubClient`, which is the interface we'll code to?

Reply:
Oh... it's because of this `p.client.PullRequests.Create` 🤦‍♂️ 

That makes my suggestion rather difficult to pull off.

Nevermind.

Reply:
Believe me, I tried 😄